agent/agent_test.go

```diff
@@ -797,6 +797,37 @@ func TestAgent_AddCheck_RestoreState(t *testing.T) {
 	}
 }
 
+func TestAgent_AddCheck_ExecDisable(t *testing.T) {
+	t.Parallel()
+
+	cfg := TestConfig()
+	cfg.CheckEnableExec = false
```

Comment:
Isn't that the default?

Update: I think this should use the normal default and tests that uses `Script` tests should set this flag to `true. The test config should use faster timings and in-mem storage but should otherwise have the same behavior than the default runtime config.